core/model/src/main/java/org/eclipse/rdf4j/model/util/ModelCollector.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.model.util;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.impl.LinkedHashModel;
+
+/**
+ * Collects a stream of Statements into a Model.
+ * 
+ * @author Bart Hanssens
+ */
+public class ModelCollector implements Collector<Statement, Model, Model> {
+	private final Object lock = new Object();
+
+	/**
+	 * Convenience method to obtain a ModelCollector.
+	 * 
+	 * @return a ModelCollector
+	 */
+	public static ModelCollector toModel() {
+		return new ModelCollector();
+	}
+
+	@Override
+	public Supplier<Model> supplier() {
+		return LinkedHashModel::new;
+	}
+
+	@Override
+	public BiConsumer<Model, Statement> accumulator() {
+		return (m, s) -> {
+			synchronized (lock) {
+				m.add(s);
+			}
+		};
+	}
+
+	@Override
+	public BinaryOperator<Model> combiner() {
+		return (m1, m2) -> {
+			m1.addAll(m2);
```

Comment:
does this also need to be synchronized?

Reply:
Actually, if I understand the Collector correctly, this method could be removed or implemented returning just null....

Because the combiner does not get called in parallelStream() with the concurrent and unordered characteristics set.
Interestingly enough, with OpenJDK11 on my machine, the combiner _does_ get called in sequential stream(), but the result seems to be ignored anyway (just returning null does not break the test code)